### PR TITLE
feat: add `sendRaw` API to Controller

### PR DESCRIPTION
**Pull request by Nerivec**
Closes https://github.com/Koenkk/zigbee-herdsman/pull/1556

* add `sendRaw` to Controller public API
  * presence of specific properties in payload determines Adapter API to use
  * enforced validation of payload properties (asserted) based on determined Adapter API
  * can pass custom clusters directly for ZCL needs (e.g. define a custom cluster in ZHC, and pass it directly here without needing it being attached to a device)
* add `disableResponse` support for InterPAN broadcast Adapter API
* cleanup Touchlink & GreenPower class a bit, provide direct `get` access instead of wrapper functions, expose a few utils to allow custom usage in ZHC
* fixes a bug with Touchlink on stop where it could have left the adapter on a different channel (probably the definition of edge case but...)

CC: @Mstrodl

TODO:
- [x] test coverage
- [x] https://github.com/Koenkk/zigbee2mqtt/pull/29730
- [x] https://github.com/Koenkk/zigbee-herdsman-converters/pull/10800

### CODE REVIEW

**Comment by Mstrodl** [comment]
Is there a way to change inter PAN channel with this? Otherwise the ability to send an inter PAN message isn't very useful

**Comment by Mstrodl** [comment]
Works for me

**Comment by Nerivec** [comment]
Had not done that part yet. Now there is. :wink: 
Let me know if you see anything across the 3 PRs.
_Once this is refined, we can add some frontend support (quick action buttons & co)._

File: src/adapter/ember/adapter/emberAdapter.ts [hunk 1]
```diff
@@ -2171,7 +2171,9 @@ export class EmberAdapter extends Adapter {
     }
 
     // queued
-    public async sendZclFrameInterPANBroadcast(zclFrame: Zcl.Frame, timeout: number): Promise<ZclPayload> {
+    public async sendZclFrameInterPANBroadcast(zclFrame: Zcl.Frame, timeout: number, disableResponse: false): Promise<ZclPayload>;
+    public async sendZclFrameInterPANBroadcast(zclFrame: Zcl.Frame, timeout: number, disableResponse: true): Promise<undefined>;
+    public async sendZclFrameInterPANBroadcast(zclFrame: Zcl.Frame, timeout: number, disableResponse: boolean): Promise<ZclPayload | undefined> {
         const command = zclFrame.command;
 
         if (command.response === undefined) {
```
Comment (Mstrodl) [inline src/adapter/ember/adapter/emberAdapter.ts]:
```suggestion
        if (!disableResponse && command.response === undefined) {
```

File: src/adapter/ezsp/adapter/ezspAdapter.ts [hunk 2]
```diff
@@ -526,16 +526,22 @@ export class EZSPAdapter extends Adapter {
...
-    public async sendZclFrameInterPANBroadcast(zclFrame: Zcl.Frame, timeout: number): Promise<ZclPayload> {
-        return await this.queue.execute<ZclPayload>(async () => {
+    public async sendZclFrameInterPANBroadcast(zclFrame: Zcl.Frame, timeout: number, disableResponse: false): Promise<ZclPayload>;
+    public async sendZclFrameInterPANBroadcast(zclFrame: Zcl.Frame, timeout: number, disableResponse: true): Promise<undefined>;
+    public async sendZclFrameInterPANBroadcast(zclFrame: Zcl.Frame, timeout: number, disableResponse: boolean): Promise<ZclPayload | undefined> {
+        return await this.queue.execute<ZclPayload | undefined>(async () => {
             logger.debug("sendZclFrameInterPANBroadcast", NS);
             const command = zclFrame.command;
 
             if (command.response === undefined) {
```
Comment (Mstrodl) [inline src/adapter/ezsp/adapter/ezspAdapter.ts]:
```suggestion
            if (!disableResponse && command.response === undefined) {
```

File: src/adapter/z-stack/adapter/zStackAdapter.ts [hunk 3]
```diff
@@ -1062,23 +1062,33 @@ export class ZStackAdapter extends Adapter {
...
+    public async sendZclFrameInterPANBroadcast(zclFrame: Zcl.Frame, timeout: number, disableResponse: false): Promise<Events.ZclPayload>;
+    public async sendZclFrameInterPANBroadcast(zclFrame: Zcl.Frame, timeout: number, disableResponse: true): Promise<undefined>;
+    public async sendZclFrameInterPANBroadcast(
+        zclFrame: Zcl.Frame,
+        timeout: number,
+        disableResponse: boolean,
+    ): Promise<Events.ZclPayload | undefined> {
+        return await this.queue.execute<Events.ZclPayload | undefined>(async () => {
             const command = zclFrame.command;
             if (command.response === undefined) {
```
Comment (Mstrodl) [inline src/adapter/z-stack/adapter/zStackAdapter.ts]:
```suggestion
            if (!disableResponse && command.response === undefined) {
```

**Review by Koenkk — APPROVED** [review]

**Comment by Koenkk** [comment]
@Nerivec not sure if it's ready, feel free to merge.

**Comment by Nerivec** [comment]
It's mostly the added sendRaw on this one. The TL/GP stuff is mostly removing unnecessary wrappers. And the extra disable response logic only touches BC InterPAN.
i.e. impact should be rather limited as far as existing code is concerned.
I think this can be merged. If anything needs tweaking, it will likely be the ZHC part.
